libfwupdplugin/fu-cabinet.c

```diff
@@ -453,6 +453,31 @@ fu_cabinet_set_container_checksum_cb (XbBuilderFixup *builder_fixup,
 	return TRUE;
 }
 
+static gboolean
+fu_cabinet_set_lowercase_checksum_cb(XbBuilderFixup *builder_fixup,
+				     XbBuilderNode *bn,
+				     gpointer user_data,
+				     GError **error)
+{
+	g_autoptr(XbBuilderNode) csum = NULL;
+
+	/* verify it exists */
+	if (g_strcmp0(xb_builder_node_get_element(bn), "artifact") == 0)
+		csum = _xb_builder_node_get_child_by_element_attr(bn, "checksum", "type", "sha256");
```

Comment:
I think you want SHA1 (and others?) too. This might mean splitting out something like `fu_cabinet_set_lowercase_checksum` as a function to use in both places.

Reply:
See if you think what I did accomplishes that (a little bit different)